Updated info about HNS vs dollarydoos

randomlogin opened the pull request:
- added information into the introduction
- added missing warnings for all endpoints involving entering the values
- split introduction into subsections (to make a reference in the warning)

pinheadmz commented on src/api-docs/source/index.html.md:
```diff
@@ -109,7 +109,18 @@ Note that if multiple user-agents are using the wallet at the same time,
-**Indexing**
+## Values
+
+Handshake currency HNS has smallest unit — one millionth of the whole, called dollarydoo. 
+<br>So <code>1 HNS = 1 000 000 dollarydoos</code>.
+<br>Different APIs require <code>value</code> and <code>rate</code> parameters nominated either in dollarydoos
+(subunits) which are integer values or in the whole HNS, which are floats.
+
+<code>value</code> and <code>rate</code> are expressed in dollarydoos when using cURL or Javascript.<br>
+<code>value</code> and <code>rate</code> are expressed in whole HNS when using CLI and RPC.
```
Great approach with the anchored links! Just to be nitpicky about the English here, can we try:


> The smallest unit of the Handshake currency (HNS) is called the "dolalrydoo" — one millionth of the whole.
> So 1 HNS = 1,000,000 dollarydoos.
> Different APIs require value and fee rate parameters denominated either in dollarydoos (subunits) which are integer values or in whole HNS, which are floats.
> 
> When using HTTP API endpoints:
> Value and rate are expressed in dollarydoos when using cURL or Javascript.
> Value and rate are expressed in whole HNS when using CLI commands.
> 
> When using RPC API commands:
> Value and rate are always expressed in whole HNS, no matter the entry method.

(and please double check that I'm correct, pretty sure that RPC always uses HNS no matter if its CLI or cURL or JS.

randomlogin commented:
If to be a bore, SI advises usage of spaces for the digit separators (1 000 000 instead of 1, 000,000) https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping .

randomlogin commented:
Sorry for doing it in the same PR. The newest changes:
- added info about standalone/plugin distinction (#85)
- added missing rpc commands (#64)
- fixed typos and missing information, for example `hsw-rpc` -> `hsw-cli rpc`

pinheadmz commented on src/api-docs/source/includes/_clients.md:
```diff
@@ -66,15 +66,17 @@ token     | _string_        | Token specific wallet
-[A sample hsd.conf file is included in the code repository](https://github.com/handshake-org/hsd/blob/master/etc/sample.conf)
-
+Command line flags should be formatted as <code>flag: true</code> in .conf files.
+<br>For example command line flag <code>--no-wallet</code> becomes <code>no-wallet: true</code> in hsd.conf.
```
This is a great explanation but actually not a good example. `--no-wallet` is NOT parsed by hsd Full Node ;-)
[Just the script in bin/node](https://github.com/handshake-org/hsd/blob/4c901f482fc21c71e1cd71d6861be8bdeceeb7af/bin/node#L40)

Any other example flag would work here like `--network=regtest` becomes `network: regtest`

randomlogin replied:
I wanted to show how flags are reformulated in the config, where the format is key: value, so the value for the flag must be 'true'. This information is not present anywhere. 

Changed example to `cors` cli flag.

I also want to add additional line to the table [here](https://hsd-dev.org/guides/config.html). I think it's also worth adding that usage of parameters as environment variables should use `export`, so it's `export HSD_NETWORK=regtest`, in that way it will be passed to the hsd process.

pinheadmz commented on src/api-docs/source/index.html.md:
```diff
@@ -123,8 +136,26 @@ along with its index in that transaction.
+Strictly speaking, any additional functionality is not part of the node itself. So the wallet is implemented as hsd node plugin. 
+
+By default hsd runs with wallet plugin activated, but it can be disabled by providing command line flag <code>--no-wallet</code>
+<br>(<code>no-wallet: true</code> in [hsd.conf](#hsd-conf-and-hsw-conf-files)).
```
mentioned earlier, command-line only

randomlogin replied:
Added. So it seems it's not possible to run hsd without wallet from the config file, right?

pinheadmz commented on src/api-docs/source/index.html.md:
```diff
@@ -123,8 +136,26 @@ along with its index in that transaction.
+Wallet runs its own server, which listens for requests.
+
+Wallet can also be run as a standalone application, connecting to local or external node:
+`wallet --node-host=127.0.0.1`
```
hm... if hsd is installed globally the command `hs-wallet` will be installed on `$PATH` which actually executes `bin/hsw`. Try it out locally and verify for me. This is related to our chat about `hsd-cli rpc` vs `hsd-rpc`... maybe this intro section should also include a mention of `npm install -g` which puts all these commands on `$PATH`:

[package.json](https://github.com/handshake-org/hsd/blob/4c901f482fc21c71e1cd71d6861be8bdeceeb7af/package.json#L55-L63)

```
  "bin": {
    "hsd": "./bin/hsd",
    "hsd-node": "./bin/node",
    "hsd-spvnode": "./bin/spvnode",
    "hs-seeder": "./bin/hs-seeder",
    "hs-wallet": "./bin/hsw",
    "hsd-cli": "./bin/hsd-cli",
    "hsw-cli": "./bin/hsw-cli"
  },
```

randomlogin replied:
Yep, global installation works, reverted back. 
I'd say we better choose one location with comprehensive info, putting everything into docs site, so it will not reference github repo page readme.

pinheadmz commented on src/api-docs/source/includes/_wallet_rpc.md:
````diff
@@ -2246,8 +2252,78 @@ N. | Name | Default |  Description
+> The above command returns JSON "result" like this:
+
+```
+552006b288266ab26fa30d9048b758a469a4101fd8235eff2384141ca5cf604d
````
That's all this RPC call returns? When I run the example I get a JSON of the entire TX.

randomlogin replied:
Added full response.

pinheadmz commented on src/api-docs/source/includes/_wallet_rpc_auctions.md:
```diff
@@ -1334,6 +1341,998 @@ address | Required | address submitting the bid
+### Params
+Name | Default |  Description
+--------- | --------- | ---------
+name | Required | name to open bidding on
```
All of these `create____` RPCs also take an optional "account" parameter.
`createopen` also accepts a totally unsused "force" bool which we have to include as a placeholder the same way "label" is required on a bunch of rpc calls even though we don't use it (but inherited that from bcoin which inherited that from Bitcoin Core...)

```
--> hsw-rpc help createopen
createopen "name" ( force "account" )
--> hsw-rpc help createbid
createbid "name" bid value ( "account" )
--> hsw-rpc help createreveal
createreveal "name" ( "account" )
--> hsw-rpc help createredeem
createredeem "name" ( "account" )
--> hsw-rpc help createupdate
createupdate "name" "data" ( "account" )
--> hsw-rpc help createrenewal
createrenewal "name" ( "account" )
--> hsw-rpc help createtransfer
createtransfer "name" "address" ( "account" )
--> hsw-rpc help createfinalize
createfinalize "name" ( "account" )
--> hsw-rpc help createcancel
createcancel "name" ( "account" )
--> hsw-rpc help createrevoke
createrevoke "name" ( "account" )
```

randomlogin replied:
Added in the description, however not using in examples for sake of simplicity, only example of `force account` for createopen is shown.

pinheadmz commented on src/api-docs/source/includes/_wallet_rpc_auctions.md:
```diff
@@ -1334,6 +1341,998 @@ address | Required | address submitting the bid
+--------- | --------- | ---------
+name | Required | name to bid on
+amount | Required | amount to bid (in HNS)
+lockup | Required | amount to lock up to blind your bid (must be greater than bid amount)
```
in the code this is actually called `value` (the value of the actual TX output). Since our RPC uses positional parameters, the names from the client perspective don't matter. But if we ever actually comply with JSON 2.0 and allow _named parameters_ (looks more like a JSON object) then we will have to change this to `value` to be correct. What are your thoughts? Come back and address it later? I know amount/lockup makes way more sense to users at this time.

randomlogin replied:
Rename "value" in the implementation to "amount".

pinheadmz replied:
gonna leave this one alone I think, it does make more sense to document it as "lockup" but we may need to revisit if we ever allow named arguments with rpc calls.

pinheadmz commented:
Also all the `create_____` should have cURL examples. I know you were probably copying the format from some other commands but those are missing cURL examples as well.

You can see examples of cURL commands in the node rpc section, like:

```
curl http://x:api-key@127.0.0.1:14037 \
  -X POST \
  --data '{
    "method": "setloglevel",
    "params": [ "none" ]
  }'
```

pinheadmz commented on src/api-docs/source/includes/_wallet_rpc_auctions.md:
````diff
@@ -1334,6 +1341,998 @@ address | Required | address submitting the bid
+```
+
+```shell--cli
+hsw-cli rpc createupdate $name '{"records": [ {"type": "NS", "ns": "ns1.example.com.", "address": "1.2.3.4"} ]}'
````
NS records dont have an `address` field anymore, this could be a GLUE4 perhaps?

randomlogin replied:
removed address field from the example

pinheadmz commented:
@randomlogin thanks so much! looks really good. I'm going to squash and merge with an extra commit just tidying up some nits: 1b5e0fd

Looks like this closes #64 #85 and #95
SO I owe you `1200 + 1200 + 1800 = 4200` HNS! I'll reach out to you for an address ;-)

THANK YOU! 🍷

pinheadmz commented:
merged: https://github.com/handshake-org/handshake-org.github.io/commit/93266f4388789d681abfeddceb8974a76c42898d
